concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -1244,8 +1248,28 @@ getAccountInfoHelper getASI getCooldowns acct bs = do
         aiAccountAddress <- BS.getAccountCanonicalAddress acc
         aiAccountCooldowns <- getCooldowns acc
         aiAccountAvailableAmount <- BS.getAccountAvailableAmount acc
-        -- TODO: Get the actual protocol layer tokens. Issue #1342
-        let aiAccountTokens = []
+        aiAccountTokens <- case sSupportsPLT (sAccountVersionFor (protocolVersion @(MPV m))) of
+            STrue -> do
+                tokenStatesMap <- BS.getAccountTokens acc
+                forM (Map.toList tokenStatesMap) $ \(tokenIndex, tokenState) -> do
+                    pltConfiguration <- BS.getTokenConfiguration bs tokenIndex
+                    let accountBalance =
+                            Tokens.TokenAmount
+                                { digits = fromIntegral $ BlockState.tasBalance tokenState,
+                                  nrDecimals = fromIntegral $ BlockState._pltDecimals pltConfiguration
+                                }
+                    return
+                        Tokens.Token
+                            { tokenId = BlockState._pltTokenId pltConfiguration,
+                              tokenAccountState =
+                                Tokens.TokenAccountState
+                                    { balance = accountBalance,
+                                      -- TODO Support allow/deny list state in account info query (Issue https://linear.app/concordium/issue/COR-1349)
+                                      memberAllowList = False,
+                                      memberDenyList = False
```

Comment:
I'm fine to leave it for now, but allow list and deny list shouldn't be part of the protobuf state, but we should have a CBOR-encoded state representation instead that includes them if relevant.